lib/reports/eligible_commitments.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+require "services"
+require "cluster"
+
+Services.mongo!
+
+module Reports
+  # Given criteria, pull up all holdings that match those criteria
+  # AND are eligible for commitments AND have no commitments.
+  # Invoke via bin/reports/compile_eligible_commitments_report.rb
+  class EligibleCommitments
+    def header
+      [
+        "organization",
+        "oclc_sym",
+        "ocn",
+        "local_id"
+      ]
+    end
+
+    def for_ocns(ocns = [])
+      if ocns.empty?
+        raise "No ocns given"
+      end
+
+      ocns.sort.uniq.each do |ocn|
+        cluster = Cluster.find_by(ocns: [ocn.to_i])
+
+        next if cluster.nil?
+        next if we_have_seen? cluster._id
+        next unless cluster.eligible_for_commitments?
```

Comment:
I'm not sure checking this up-front helps that much since we end up needing to iterate over holdings to determine this anyway.